File: SpringBootApp/src/main/java/com/example/springbootapp/data/entities/Embdeddables/Personalization.java
package com.example.springbootapp.data.entities.Embdeddables;

import jakarta.persistence.Embeddable;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.Objects;

@Embeddable
@Data
@NoArgsConstructor
public class Personalization {

    private static final Double PLAYER_NAME_PRICE = 10.0;
    private static final Double PLAYER_NUMBER_PRICE = 5.0;
    private String playerName;
    private Integer playerNumber;

    public double getPrice() {
        double price = 0.0;
        if (playerName != null && !playerName.isEmpty()) {
            price += PLAYER_NAME_PRICE;
        }
        if (playerNumber != null) {
            price += PLAYER_NUMBER_PRICE;
        }
        return price;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Personalization that = (Personalization) o;
        return playerName.equals(that.playerName) &&
                playerNumber.equals(that.playerNumber);
    }

    @Override
    public int hashCode() {
        return Objects.hash(playerName, playerNumber);
    }
}

File: ClientAdmin/app/src/main/java/com/example/clientadmin/model/Personalization.kt
package com.example.clientadmin.model

class Personalization(
    val playerName: String,
    val playerNumber: Int
) {
    companion object{
        val PLAYER_NAME_PRICE: Double = 10.0
        val PLAYER_NUMBER_PRICE: Double = 5.0
    }


    //modificate le condizioni, da verificare se vanno bene
    fun getPrice(): Double{
        var price = 0.0
        if (playerName != "") {
            price += PLAYER_NAME_PRICE
        }
        if (playerNumber > 0) {
            price += PLAYER_NUMBER_PRICE
        }
        return price
    }
}